modules/options/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -183,8 +183,8 @@ final case class BuildOptions(
     javaOptions.javaHomeOpt
       .orElse {
         if (javaOptions.jvmIdOpt.isEmpty)
-          sys.props.get("java.home").map(p =>
-            Positioned(Position.Custom("java.home prop"), os.Path(p, Os.pwd))
+          Option(System.getenv("JAVA_HOME")).map(p =>
```

Comment:
Let's keep a fallback to `sys.props.get("java.home")`, if `JAVA_HOME` isn't set. On the JVM, this property should be set even when `JAVA_HOME` isn't (like if the `bin` directory is in the `PATH` and `JAVA_HOME` isn't set).

Reply:
Right, that makes sense